feat: support nested project variables (#1220)
canonical/craft-parts: docs/reference/changelog.rst

  dependency versions available on the host, aligning with how Maven normally behaves.
  This change makes the plugin a drop-in replacement for Maven in private networks.

> - Adds support for nested project variables. Nested project variables are managed with
>   craftctl using dot notation. For example, ``craftctl set var.subvar=foo`` sets the
>   nested project variable ``var.subvar`` to ``foo``.

Reviewer: ```suggestion
- Add support for nested project variables, which can be referenced with
  craftctl using dot notation. For example, ``craftctl set var.subvar=foo`` sets the
  nested project variable ``var.subvar`` to ``foo``.
```

Author: https://github.com/canonical/craft-parts/pull/1220/commits/c76340c421d68e72fa0ae0d746f8ed42f1dfd178
